astropy/units/imperial.py

```diff
@@ -147,6 +147,7 @@
 )
 def_unit(
     ["deg_R", "Rankine"],
+    CompositeUnit(5 / 9, [si.K], [1]),
```

Comment:
In analogy with the other definitions, and for readability, let's create the composit unit implicitly, via `5 / 9 * si.K`.

Reply:
According to the documentation of `def_unit()` the `represents` argument must be a `UnitBase` instance: https://github.com/astropy/astropy/blob/d816139637181c4f3f1997bc46c775e7f2ea4d02/astropy/units/core.py#L2616
The change you proposed would mean changing the `represents` argument here from a `UnitBase` to a `Quantity`. Although it is true that this would (currently) also work, we should avoid writing new code that relies on undocumented implementation details like that.

Reply:
That may be so, but it is literally used in every other unit definition!

Reply:
I think it's sufficiently useful to keep it that users can call `def_unit` with a Quantity argument. And of course we should have examples to this effect. But I'm not sure we should be doing it in our internal `def_unit` calls. I like this change, particularly if it's effectuated throughout.

Reply:
This is not a question of style. Documentation defines the API, so as long as `def_unit()` says that `represents` must be a `UnitBase`, there can be no expectation that passing in a `Quantity` will work.

Reply:
This is good! We're encountering the difference between documentation, typing, and usage. They should all reflect the same API and we should decide what that is.

Reply:
I agree the use of CompositeUnit is unwieldy, but I really don't think inside Astropy we should be building units from Quantities. There's a hierarchy of dimensions to units to quantities and it should only build up, not down.

Reply:
Please let's have this discussion in a separate issue. There is no need to break consistency with everything else here.

And, yes, clearly the documentation is wrong, which should be corrected.

Reply:
`def_unit()` documentation was updated in a separate pull request, so we are not forced to explicitly create a `CompositeUnit` anymore.